server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -407,6 +407,60 @@ private TableMgmtStats manageTablets(Iterator<TabletManagement> iter,
             state, goal, actions);
       }
 
+      // If the current tablet is hosted and the TabletServer that it's hosted on is being shut
+      // down, then we need to unhost tablets in order. We can't unassign the root tablet before
+      // moving any metadata table tablets that are on this host. We can't unassign metadata
+      // tablets before moving any user table tablets that are on this host.
+      if (current != null && state == TabletState.HOSTED
```

Comment:
If a tablet server is hosting metadata and user tablet, it should be ok to unload the metadata tablet first if there is another tablet server that can host it.  Once the metadata tablet is loaded elsewhere the the user tablet can unload.  If all tservers are being shutdown, then it seems like the existing code should handle this.

Reply:
Yeah, I wasn't sure about the user tables, but I followed the same pattern. For sure, the metadata tablets have to be moved **before** the root tablet. I'll try your suggestion regarding the user tablets.

Reply:
I am trying to understand the use case for the new code in this PR.  As long as there is another tablet server to load root or metadata tablet, then it should not matter what order they are unloaded in.  It only matters when all tablet servers are being shutdown, which I thought there was existing code to handle.